spicy/toolchain/src/ast/types/unit-items/field.cc

```diff
@@ -30,3 +33,30 @@ std::optional<std::pair<const Expression, std::optional<const Type>>> spicy::typ
 
     return {};
 }
+
+std::optional<const Expression> spicy::type::unit::item::Field::size() const {
```

Comment:
Couple things:

Good point that the size can depend not just on types, but also on field attributes, I didn't think about that. Given that, I'm now wondering if the type-side `size()` method is really as helpful as I originally thought because we now have two levels of `size()`, each with just a couple of cases/implementations (`Type::size()` and (`Field::size`). So maybe putting it all here into `Field::size()` is the better approach.

And then I think I'd use a visitor instead of the if/else-chain, both for readability/consistency-in-style and (I believe ...) performance.

Reply:
I changed this as suggested.